controller/loginController.js

```diff
@@ -0,0 +1,113 @@
+const User = require('../models/user')
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcrypt')
+const http = require('http-status-codes')
+const secretKey = require('../config/secretKey')
+const statusMsg = require('../config/statusMsg')
+const dotenv = require('dotenv')
+dotenv.config({
+    path: './config/.env'
+})
+const tokenGenerator = require('./authTokenGenerator')
+
+exports.login = async (req, res, next) => {
+    try {
+        if (req.body.email === '' || req.body.password === '') {
+            let err = new Error()
+            err.status = http.CONFLICT
+            next(err)
+        }
+        let user = await User.findOne({ email: req.body.email })
+        if (!user) {
+            let err = new Error()
+            err.status = http.BAD_REQUEST
+            return next(err)
+        }
+        let result = await bcrypt.compare(req.body.password, user.password)
+        if (!result) {
+            return res.status(http.BAD_REQUEST).json({
+                "success": statusMsg.fail.msg,
+                "payload": { email: req.body.email },
+                "error": {
+                    "code": http.BAD_REQUEST,
+                    "message": statusMsg.password_not_match.msg
+                }
+            })
+        }
+        if (user.verified_email && user.password) {
```

Comment:
@AwaleRohin do you need to check ```user.password```?

Reply:
yes, because if user signup with facebook then email will be verified but there will be no password